iothub/service/src/Configurations/ConfigurationsClient.cs

```diff
@@ -0,0 +1,389 @@
+                if (!string.IsNullOrEmpty(configuration.ETag))
+                {
+                    throw new ArgumentException(ApiResources.ETagSetWhileCreatingConfiguration);
+                }
+                cancellationToken.ThrowIfCancellationRequested();
+
+                using HttpRequestMessage request = _httpRequestMessageFactory.CreateRequest(HttpMethod.Put, GetConfigurationRequestUri(configuration.Id), _credentialProvider, configuration);
+                HttpResponseMessage response = await _httpClient.SendAsync(request, cancellationToken);
+                await HttpMessageHelper2.ValidateHttpResponseStatus(HttpStatusCode.OK, response);
+                return await HttpMessageHelper2.DeserializeResponse<Configuration>(response, cancellationToken);
+            }
+            catch (Exception ex)
+            {
+                if (Logging.IsEnabled)
+                    Logging.Error(this, $"{nameof(CreateAsync)} threw an exception: {ex}", nameof(CreateAsync));
+                throw;
+            }
+            finally
+            {
+                if (Logging.IsEnabled)
+                    Logging.Exit(this, $"Creating configuration: {configuration?.Id}", nameof(CreateAsync));
+            }
+        }
+
+        /// <summary>
+        /// Retrieves the specified configuration object.
+        /// </summary>
+        /// <param name="configurationId">The id of the configuration being retrieved.</param>
+        /// <param name="cancellationToken">The token which allows the operation to be canceled.</param>
+        /// <returns>The Configuration object.</returns>
+        /// <exception cref="ArgumentNullException">Thrown when the provided configuration Id is null.</exception>
+        /// <exception cref="IotHubException">
+        /// Thrown if IoT hub responded to the request with a non-successful status code. For example, if the provided
+        /// request was throttled, <see cref="IotHubThrottledException"/> is thrown. For a complete list of possible
+        /// error cases, see <see cref="Common.Exceptions"/>.
+        /// </exception>
+        /// <exception cref="HttpRequestException">
+        /// If the HTTP request fails due to an underlying issue such as network connectivity, DNS failure, or server
+        /// certificate validation.
+        /// </exception>
+        /// <exception cref="OperationCanceledException">If the provided cancellation token has requested cancellation.</exception>
+        /// <seealso href="https://docs.microsoft.com/azure/iot-hub/iot-hub-automatic-device-management"/>
+        public async virtual Task<Configuration> GetAsync(string configurationId, CancellationToken cancellationToken = default)
+        {
+            if (Logging.IsEnabled)
+                Logging.Enter(this, $"Getting configuration: {configurationId}", nameof(GetAsync));
+            try
+            {
+                Argument.RequireNotNull(configurationId, nameof(configurationId));
```

Comment:
Should be checking if not null **or empty**
```suggestion
                Argument.RequireNotNullOrEmpty(configurationId, nameof(configurationId));
```

You also need your xml header to mention that ArgumentException is thrown when configurationId is empty or whitespace